### PR TITLE
go-to-protobuf: fix rewrite of embedded struct fields

### PR DESCRIPTION
**Pull request by mandelsoft**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change

/kind bug
/priority important-soon

> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

The protobuf generator cannot handle embedded struct fields correctly.
When using an embedded field by pointer it produces the error message

`unable to get name for tag from struct "...", field ...`

The reason is that the name determination evaluating the AST does not
handle pointers if the AST does not already contain a field name.
This seems to be the case for structs embedded by pointers.

Example:

```
type MyStruct struct {
  *OtherStruct `protobuf:"bytes,1,opt,name=otherStruct"`
}
```
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by rfranzke** [comment]
@krousey @jennybuckley @sttts @smarterclayton Do you have some time to check this PR/provide some feedback?

**Comment by smarterclayton** [comment]
What is the use case for an embedded struct pointer?

**Comment by mandelsoft** [comment]
@smarterclayton We want to extend an API in a compatible manner. In the concrete example we have a `*LabelSelector` field in the spec of  the previous version and want to replace it with a more detailed type (`*SeedSelector`), which now has con include the embedded type `*LabelSelector` to be compatible.

**Comment by rfranzke** [comment]
/unassign @krousey @sttts @jennybuckley
/assign @lavalamp @wojtek-t

according to https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md#quirks-of-the-process

**Comment by lavalamp** [comment]
/ok-to-test

Please add a test.

**Comment by mandelsoft** [comment]
I cannot find any real comprehensive tests for this package, so I've just added a test for the new function used to determine the name of a struct field.

File: staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/parser_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (lavalamp) [inline staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/parser_test.go:2]: 2020

File: staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/parser_test.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	"testing"
+)
+
+func TestProtoParser(t *testing.T) {
+	ident := ast.NewIdent("FieldName")
+	tests := []struct {
+		expr ast.Expr
+		err  bool
+	}{
+		{
```
Comment (lavalamp) [inline staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/parser_test.go:30]: I'm nitpicking now, but a snippet (e.g. in a comment) of go code that produces the AST snippet would be helpful to readers?

**Comment by rfranzke** [comment]
ping @lavalamp

**Comment by lavalamp** [comment]
I just realized this PR checks in a binary. Sorry, I should have noticed that earlier. Can you send a follow up to remove it? And update the cherry-picks?

**Pull request merged**